src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -306,15 +314,40 @@ namespace AppInstaller::CLI::Workflow
 
             bool IsFirstBetter(const Manifest::ManifestInstaller& first, const Manifest::ManifestInstaller& second) override
             {
-                // TODO: Current implementation assumes there is only a single installer type requirement. This needs to be updated
-                // once multiple installerType requirements and preferences are accepted.
-                UNREFERENCED_PARAMETER(first);
-                UNREFERENCED_PARAMETER(second);
-                return true;
+                if (m_preference.empty())
+                {
+                    return false;
+                }
+
+                InstallerTypeEnum firstBaseInstallerType = first.BaseInstallerType;
+                InstallerTypeEnum firstEffectiveInstallerType = first.EffectiveInstallerType();
+
+                InstallerTypeEnum secondBaseInstallerType = second.BaseInstallerType;
+                InstallerTypeEnum secondEffectiveInstallerType = second.EffectiveInstallerType();
+
+                for (auto const& preferredInstallerType : m_preference)
+                {
+                    bool isFirstInstallerTypePreferred = (preferredInstallerType == firstBaseInstallerType) || (preferredInstallerType == firstEffectiveInstallerType);
+                    bool isSecondInstallerTypePreferred = (preferredInstallerType == secondBaseInstallerType) || (preferredInstallerType == secondEffectiveInstallerType);
+
+                    if (isFirstInstallerTypePreferred && isSecondInstallerTypePreferred)
+                    {
+                        return false;

```

Comment:
The more I think about this, I'm not sure the for loop is best here?

Presume for a moment `m_preference` is `{exe, nullsoft, inno}` - `first.BaseInstallerType` is `exe` - `second.BaseInstallerType` is `inno`

The very first time through the for loop, `preferredInstallerType` will be exe. When it goes through the comparison, it will return `true` indicating that the exe is better than the inno.

-----

This is good if the preferences list is intended to be in order of preference. But, if it's intended to be that all preferences are of equal preference, then this won't be accurate - we would need a member function to check if the preferences vector contains the installer type.
```c++
bool PreferenceContains(InstallerTypeEnum installerType) {
    return std::find(m_preference.begin(), m_preference.end(), installerType) != m_preference.end();
}
```

The other thing that may need to be considered is how this interacts with the other applicability filters. 
On an x64 system - if there is an `x86 - exe` and and `x64 - inno`, which would be selected? The architecture filter would prefer the x64, but if the user has specified preferences of `{exe, nullsoft, inno}` , would it still select the Exe as preferred since it's first in the list?

Reply:
Looked into the code further - Architecture preferences/requirements are handled where any architecture in the list is allowed, and order is not considered for preference. I'd suggest we use the same approach here

Reply:
You bring up some great points, all of which I agree with. I had originally thought that order of preference would be a good idea, but this implementation implies that there is *always* an order and I think that could introduce an unnecessary behavior. A user would have no way of declaring preferences that are all equally preferred.

I also think that if other setting preferences don't have this notion of an order, I should keep the behavior the same and not introduce that just for installer types. 

I have updated preference to simply check if the installer type exists in the preferences and determining that result based on that.